mlflow/dspy/callback.py

```diff
@@ -204,6 +215,80 @@ def on_tool_end(
         if call_id in self._call_id_to_span:
             self._end_span(call_id, outputs, exception)
 
+    def on_evaluate_start(self, call_id: str, instance: Any, inputs: dict[str, Any]):
+        """
+        Callback handler at the beginning of evaluation call. Available with DSPy>=2.6.9.
+        This callback starts a nested run for each evaluation call if called within optimization,
+        and created a new run when called outside of optimization and there is no active run.
+        """
+        if not get_autologging_config(FLAVOR_NAME, "log_evals"):
+            return
+        if devset := inputs.get("devset"):
+            is_minibatch = len(devset) < len(instance.devset)
```

Comment:
not blocking this PR, but this check could cause issues if users don't carefully configure their Evaluator class, since `devset` at call time override the `self.devset`: https://github.com/stanfordnlp/dspy/blob/112d7115c21899502a6a1288d540bf49da77980e/dspy/evaluate/evaluate.py#L135

Reply:
This logic will be removed since we will receive key